File: api/src/main/models/invite.py
from sqlalchemy import Column, ForeignKey, Integer, String
from sqlalchemy.orm import relationship
from src.main.database import Base


class Invite(Base):
    __tablename__ = "invites"
    id = Column(Integer, primary_key=True, index=True)
    event_id = Column(
        Integer, ForeignKey("events.id", ondelete="CASCADE"), nullable=False
    )
    user_id = Column(
        Integer, ForeignKey("users.id", ondelete="CASCADE"), nullable=True
    )
    email = Column(String, nullable=False)
    role = Column(String, nullable=False)
    token = Column(String, unique=True, nullable=False)
    status = Column(String, nullable=False, default="pending")
    event = relationship("Event", back_populates="invites")
    user = relationship("User", back_populates="invites")

File: ui/src/types/invite.ts
import { EventOut } from './event';

export interface InviteBase {
    email: string;
    role: string;
}

export interface InviteCreate extends InviteBase {
    eventId: number;
}

export interface InviteOut extends InviteBase {
    event: EventOut;
    id: number;
    status: string;
    token: string;
    userName: string;
}

export interface InviteStatusUpdate {
    status: string;
}
